javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -34,11 +37,19 @@ public void beforeAgent(Config config) {
   @SuppressWarnings("unused")
   public static synchronized void installAgentTracer(Config config) {
     if (config.getBooleanProperty(JAVAAGENT_ENABLED_CONFIG, true)) {
+
       copySystemProperties(config);
 
-      OpenTelemetrySdk sdk = OpenTelemetrySdkAutoConfiguration.initialize();
-      OpenTelemetrySdkAccess.internalSetForceFlush(
-          (timeout, unit) -> sdk.getSdkTracerProvider().forceFlush().join(timeout, unit));
+      if (config.getBooleanProperty(JAVAAGENT_NOOP_CONFIG, false)) {
+        GlobalOpenTelemetry.set(NoopOpenTelemetry.getInstance());
+      } else {
+        System.setProperty("io.opentelemetry.context.contextStorageProvider", "default");
```

Comment:
Please create a new issue about this unwanted side effect.

Reply:
Created #3566 :)